vulture/config.py

```diff
@@ -0,0 +1,184 @@
+"""
+This module handles retrieval of configuration values from either the
+command-line arguments or the pyproject.toml file.
+"""
+from __future__ import print_function
+import argparse
+import sys
+from os.path import abspath, exists
+from typing import Any, Dict
+
+import toml
+
+from .version import __version__
+
+
+class Config(Dict[str, Any]):
+    """
+    A config object wrapping various variables and used to abstract away the
+    difference between CLI-arg parsing and TOML loading.
+    """
+
+    #: Possible configuration options and their respective defaults
+    DEFAULTS = {
+        "min_confidence": 0,
+        "paths": [],
+        "exclude": [],
+        "ignore_decorators": [],
+        "ignore_names": [],
+        "make_whitelist": False,
+        "sort_by_size": False,
+        "verbose": False,
+    }
+
+    def __getattribute__(self, name):
+        if name in self:
+            return self[name]
+        return super(Config, self).__getattribute__(name)
+
+    def __repr__(self):
+        return "Config(%r)" % dict(self)
```

Comment:
Can you use str.format() for new code please?

Reply:
Addressed in 9811bb4